apps/emqx_resource/src/schema/emqx_resource_schema.erl

```diff
@@ -47,6 +47,7 @@ fields("resource_opts") ->
 fields("creation_opts") ->
     create_opts([]).
 
+-spec create_opts([{atom(), hocon_schema:field_schema_map()}]) -> [{atom(), term()}].
```

Comment:
```suggestion
-spec create_opts([{atom(), hocon_schema:field_schema()}]) -> [{atom(), hocon_schema:field_schema()}].
```

or maybe 
```suggestion
-spec create_opts([hocon_schema:field()]) -> [hocon_schema:field()].
```

Reply:
`hocon_schema:override` specifically accepts `field_schema_map()`, and `field_schema()` encompasses it plus other types, so I think it might be misleading.  `field()` also seems slightly different from `field_schema_map()`.  :thinking:

Reply:
I meant for the function argument, though.  For the result type, it indeed matches.

Reply:
:heavy_check_mark: